### DISCUSSION TITLE
gnet.conn使用疑问

### DISCUSSION THREAD

**Discussion by Vanlius** [panjf2000/gnet#294] [Q&A]
1. 我不太理解这里的Context接口函数，说明是返回一个用户自定义上下文，这个上下文主要是干什么的？一般用在什么场景？
https://github.com/panjf2000/gnet/blob/51fb73842710051c216dbea9c170d62e7eafcf4b/gnet.go#L95

2. ICodec的作用是在发包和解包的时候，直接序列化和反序列化指定的类型吗？ 
https://github.com/panjf2000/gnet/blob/51fb73842710051c216dbea9c170d62e7eafcf4b/codec.go#L31

3. ReadN()函数是读取到byte到指定的位置获取[]byte数据，那有没有我只读取指定区间位置（比如32-36）的[]byte呢？

**Comment by panjf2000** [comment]
> 我不太理解这里的Context接口函数，说明是返回一个用户自定义上下文，这个上下文主要是干什么的？一般用在什么场景？

Context 是 Conn 用于在生命周期内的传递的信息。

> ICodec的作用是在发包和解包的时候，直接序列化和反序列化指定的类型吗？

用于分包和组包，序列化和反序列化目前要在 React() 里做。

> ReadN()函数是读取到byte到指定的位置获取[]byte数据，那有没有我只读取指定区间位置（比如32-36）的[]byte呢？

没有。
